ecl/provider_connectivity/v2/tenant_connections/requests.go

```diff
@@ -0,0 +1,165 @@
+package tenant_connections
+
+import (
+	"github.com/nttcom/eclcloud"
+	"github.com/nttcom/eclcloud/pagination"
+)
+
+// ListOptsBuilder allows extensions to add additional parameters to
+// the List request
+type ListOptsBuilder interface {
+	ToTenantConnectionListQuery() (string, error)
+}
+
+// ListOpts provides options to filter the List results.
+type ListOpts struct {
+	TenantConnectionRequestID string `q:"tenant_connection_request_id"`
+	Status                    string `q:"status"`
+	Name                      string `q:"name"`
+	TenantID                  string `q:"tenant_id"`
+	NameOther                 string `q:"name_other"`
+	TenantIDOther             string `q:"tenant_id_other"`
+	NetworkID                 string `q:"network_id"`
+	DeviceType                string `q:"device_type"`
+	DeviceID                  string `q:"device_id"`
+	DeviceInterfaceID         string `q:"device_interface_id"`
+	PortID                    string `q:"port_id"`
+}
+
+// ToTenantConnectionListQuery formats a ListOpts into a query string.
+func (opts ListOpts) ToTenantConnectionListQuery() (string, error) {
+	q, err := eclcloud.BuildQueryString(opts)
+	if err != nil {
+		return "", err
+	}
+	return q.String(), nil
+}
+
+// List retrieves a list of Tenant Connection.
+func List(client *eclcloud.ServiceClient, opts ListOptsBuilder) pagination.Pager {
+	url := listURL(client)
+	if opts != nil {
+		query, err := opts.ToTenantConnectionListQuery()
+		if err != nil {
+			return pagination.Pager{Err: err}
+		}
+		url += query
+	}
+	return pagination.NewPager(client, url, func(r pagination.PageResult) pagination.Page {
+		return TenantConnectionPage{pagination.LinkedPageBase{PageResult: r}}
+	})
+}
+
+// Get retrieves details of an Tenant Connection.
+func Get(client *eclcloud.ServiceClient, id string) (r GetResult) {
+	_, r.Err = client.Get(getURL(client, id), &r.Body, nil)
+	return
+}
+
+// CreateOptsBuilder allows extensions to add additional parameters to
+// the Create request.
+type CreateOptsBuilder interface {
+	ToTenantConnectionCreateMap() (map[string]interface{}, error)
+}
+
+// ServerFixedIPs contains the IP Address and the SubnetID.
+type ServerFixedIPs struct {
+	SubnetID  string `json:"subnet_id,omitempty"`
+	IPAddress string `json:"ip_address,omitempty"`
+}
+
+// AddressPair contains the IP Address and the MAC address.
+type AddressPair struct {
+	IPAddress  string `json:"ip_address,omitempty"`
+	MACAddress string `json:"mac_address,omitempty"`
+}
+
+// VnaFixedIPs represents ip address part of virtual network appliance.
+type VnaFixedIPs struct {
+	IPAddress string `json:"ip_address,omitempty"`
+}
+
+// Vna represents the parameter when device_type is VSRX.
+type Vna struct {
+	FixedIPs []VnaFixedIPs `json:"fixed_ips,omitempty"`
+}
+
+// Server represents the parameter when device_type is Server.
+type Server struct {
```

Comment:
`SegmentationID` and `SegmentationType` can be set with only bare metal servers. So can you separate it to bare metal and compute.
https://ecl.ntt.com/en/documents/api-references/provider-connectivity/rsts/APIList/source/v2.0/TC/create_tenant_connection.html#id7

Reply:
understood. I'll try.

Reply:
Thanks. Can you add test case for create bare metal one?

Reply:
It is certainly necessary.Add that too.